src/services/database.service.ts

```
@@ -127,6 +127,173 @@ export class DatabaseService {
+  async saveUserProfileDataWithDynamicFields(data: Record<string, any>) {
+    const meta = this.userRepo.metadata;
+
+    // Collect all property names known to TypeORM (from the entity)
+    const knownPropertyNames = new Set(meta.columns.map((c) => c.propertyName));
+
+    // Exact SCREAMING_SNAKE label for each dynamic camelKey, supplied by
+    // TransformService. Using this instead of reverse-deriving the label from
+    // the camelKey avoids mismatches for labels with digits/punctuation/mixed
+    // case, which would otherwise bypass HANDLED_LABEL_TO_COLUMN and create a
+    // duplicate column for a field that was already mapped.
+    const dynamicFieldLabels: Record<string, string> = data._dynamicFieldLabels || {};
```

ESLint

[error] Insert `⏎·····`

(prettier/prettier)